src/components/SearchBar/SearchInput.tsx

```diff
@@ -58,11 +61,33 @@ export const SearchInput = forwardRef<ISearchInputProps, 'input'>((props, ref) =
   const [input, focus] = useFocus({ selectTextOnFocus: false });
   const refs = useMergeRefs(ref, input);
   const { query, queryAddition, onDoneAppendingToQuery, isClearingQuery, onDoneClearingQuery } = useIntermediateQuery();
+  const [userInput, setUserInput] = useState<{ value: string; cursorPos: number }>({ value: '', cursorPos: 0 });
+  const deferredUserInput = useDeferredValue(userInput);
+  const [uatSearchTerm, setUatSearchTerm] = useState(null);
+
+  const { data: uatSearchTermData } = useUATTermsSearch({ term: uatSearchTerm }, { enabled: !!uatSearchTerm });
+
+  useEffect(() => {
+    if (!!uatSearchTermData) {
+      // convert to typeahead options and dispatch
+      const options = uatSearchTermData.uatTerms.map((r, i) => {
+        return {
+          value: `"${r.name}"`,
+          label: r.name,
+          desc: [r.name, ...(r.altNames || [])].join(', '),
+          id: i,
+          match: [] as string[],
+        } as TypeaheadOption;
+      });
```

Comment:
I wonder if this could be rolled into the API response so it gets cached, since you have to do it at some point, this would allow you to skip it for subsequent requests.

Reply:
Initially I was doing just what you described. After thinking a bit more, I felt the API should not return something so specific, maybe later on it can be used in other components.